geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/index/s2/S2IndexKeySpace.scala

```diff
@@ -164,14 +168,14 @@ class S2IndexKeySpace(val sft: SimpleFeatureType, val sharding: ShardStrategy, g
   override def getRangeBytes(ranges: Iterator[ScanRange[Long]], tier: Boolean): Iterator[api.ByteRange] = {
     if (sharding.length == 0) {
       ranges.map {
-        case BoundedRange(lo, hi) => BoundedByteRange(ByteArrays.toBytes(lo), ByteArrays.toBytes(hi))
+        case BoundedRange(lo, hi) => BoundedByteRange(ByteArrays.toBytes(lo), ByteArrays.toBytesFollowingPrefix(hi))
         case r => throw new IllegalArgumentException(s"Unexpected range type $r")
       }
     } else {
       ranges.flatMap {
         case BoundedRange(lo, hi) =>
           val lower = ByteArrays.toBytes(lo)
-          val upper = ByteArrays.toBytes(hi)
+          val upper = ByteArrays.toBytesFollowingPrefix(hi)
```

Comment:
Is this an albeit-small change in logic?  If so, can we cover it with a unit test?

Reply:
our ranges are supposed to be inclusive starts and exclusive ends, so this was a bug fix. the issue actually popped up b/c the hbase unit tests were failing due to the start and end being the same in some cases.